### ISSUE TITLE
PEP 585 support is broken again

### ISSUE THREAD

**Issue by Tshimanga** [python/mypy#17552]
**Bug Report**

#9873 is back.

**To Reproduce**

```python
foo: list[int] = [1,2,3]
```

**Expected Behavior**

No Complaints.

**Actual Behavior**

```
error: "list" is not subscriptable, use "typing.List" instead  [misc]
```

**Your Environment**

- Mypy version used: 1.11.0
- Python version used: 3.11 & 3.12

**Comment by asottile-sentry** [comment]
works fine for me -- I suspect you're running mypy in an older python version and/or with a configuration setting selecting an older python:

```console
$ mypy -c 'foo: list[int] = [1,2,3]'
Success: no issues found in 1 source file
$ mypy --version && python --version
mypy 1.11.0 (compiled: yes)
Python 3.12.2

**Comment by Tshimanga** [comment]
> I suspect you're running mypy in an older python version
No, I am running mypy 1.11.0 and python 3.12 (as I indicated in my environment details above)

**Comment by JukkaL** [comment]
@Tshimanga Are you still experiencing the issue? Can you try creating a fresh venv using 3.9 or later, create an empty directory, and run `mypy -c 'foo: list[int] = [1,2,3]'`, `mypy --version` and `python --version` in it? Can you also try running `mypy --python-version 3.9 -c 'foo: list[int] = [1,2,3]'`?
